### PR TITLE
Course managment

### PR DESCRIPTION
**Pull request by Isharam99**

### CODE REVIEW

File: models/questionModel.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+   
+    
+    isApproved:{ // for admin
+        type: Boolean,
+        default: true,
+    }
+   
+},{timestamps :true}) 
+
+module.exports =  mongoose.model("Question",QuestionSchema)
```
Comment (Ushan-Dilusha) [inline models/questionModel.js]:
```suggestion
module.exports =  mongoose.model("Question",QuestionSchema)    

```

File: models/quizSetModel [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    },
+    quizzes: [
+      {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: 'Quiz', // Reference to the Quiz model
+      },
+    ],
+  }, { timestamps: true });
+  
+  module.exports = mongoose.model("QuizSet", QuizSetSchema);
```
Comment (Ushan-Dilusha) [inline models/quizSetModel]:
```suggestion
  module.exports = mongoose.model("QuizSet", QuizSetSchema);
  
```

File: uploads/IT21183904_KUMARA_H.M.I.M.jpg
Comment (Ushan-Dilusha) [inline uploads/IT21183904_KUMARA_H.M.I.M.jpg:1]: what is this

File: controllers/quizCtrl.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      if (!quizzes || quizzes.length === 0) {
+        return res.status(404).json({ error: 'No quizzes found for the specified type' });
+      }
+  
+      res.status(200).json({ quizzes });
+    } catch (error) {
+      console.error('Error filtering quizzes by type:', error);
+      res.status(500).json({ error: 'An error occurred while filtering quizzes by type.' });
+    }
+  };
```
Comment (Ushan-Dilusha) [inline controllers/quizCtrl.js]:
```suggestion
  };
  
```

File: controllers/questionCtrl.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    }
+
+    res.status(200).json({ quizSet });
+  } catch (error) {
+    console.error('Error getting quiz set by ID:', error);
+    res.status(500).json({ error: 'An error occurred while fetching the quiz set.' });
+  }
+};
+  
+
```
Comment (Ushan-Dilusha) [inline controllers/questionCtrl.js]: remove unwanted lines

**Review by Ushan-Dilusha — CHANGES_REQUESTED** [review]

File: controllers/questionCtrl.js (hunk 4 above)
Comment (Isharam99) [reply controllers/questionCtrl.js]: fixed https://github.com/Ushan-Dilusha/mysticwebcraft-backend/pull/14/commits/e20fe387cf1942b54c2c22b65613fec770943005

**Review by Ushan-Dilusha — APPROVED** [review]

**Pull request merged**